float32 a_joybutton
float32 b_joybutton
float32 x_joybutton
float32 y_joybutton